### PR TITLE
Logs diagnostics about sessions without ID.me UUID

### PR DESCRIPTION
**Pull request by patrickvinograd**
## Description of change
We want to consider consuming ID.me UUIDs from the accounts table but first are going to collect data about what percentage of inbound users this would help. 

## Things to know about this PR
No PII concerns, data is logged to cloudwatch and contains no user-specific identifiers.

### CODE REVIEW

File: app/controllers/v1/sessions_controller.rb [hunk 1]
```diff
@@ -175,6 +176,15 @@ def user_logout(saml_response)
       end
     end
 
+    # Diagnostic logging to determine what percentage of these issues
+    # would be resolved by an account lookup, before we implement that
+    def log_missing_uuid_info(exception)
+      return if exception&.identifier.blank?
+      accounts = Account.where(icn: exception.identifier)
```
Comment (ericbuckley) [inline app/controllers/v1/sessions_controller.rb:183]: would attempting to match by icn OR secid increase our odds?
```
accounts = Account.where(icn: exception.attributes[:mhv_icn]).or(Account.where(sec_id: exception.attributes[:sec_id])
```

File: lib/saml/errors.rb [hunk 2]
```diff
@@ -19,11 +19,14 @@ class UserAttributeError < SAMLError
...
 
-    def initialize(message:, code:, tag:)
+    attr_reader :identifier
+
+    def initialize(message:, code:, tag:, identifier: nil)
       @code = code
       @tag = tag
       @level = :warning
       @context = {}
+      @identifier = identifier
```
Comment (ericbuckley) [inline lib/saml/errors.rb:29]: Do we need to filter this in sentry?

File: app/controllers/v1/sessions_controller.rb (hunk 1 above)
Comment (patrickvinograd) [reply app/controllers/v1/sessions_controller.rb:183]: Technically yes but I just checked production and there are 12 accounts with a secid but no ICN. Whereas there are 2,192,808 accounts with an ICN.

File: lib/saml/errors.rb (hunk 2 above)
Comment (patrickvinograd) [reply lib/saml/errors.rb:29]: I don't think so; ICN is fair game to log, like ID.me UUID and secid. But we should go through and bring https://github.com/department-of-veterans-affairs/vets-api/pull/4095 to completion.

File: app/controllers/v1/sessions_controller.rb (hunk 1 above)
Comment (ericbuckley) [reply app/controllers/v1/sessions_controller.rb:184]: I was thinking about the case where the secid would be in the SAML Response, but not the mhv_icn.

**Pull request merged**